addons/website_payment_stripe/controllers/main.py

```diff
@@ -0,0 +1,19 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+
+from odoo.addons.payment_stripe.controllers.main import StripeController
+from odoo.modules.module import get_resource_path
+
+
+class WebsiteStripeController(StripeController):
+    _apple_pay_url = '/.well-known/apple-developer-merchantid-domain-association'
+
+    @http.route(_apple_pay_url, type='http', auth='public', csrf=False)
+    def stripe_apple_pay(self):
+        """ TODO VCR
+        """
+        file_path = get_resource_path(
+            'website_payment_stripe', 'static', 'apple-developer-merchantid-domain-association'
+        )
+        return open(file_path, 'rb').read()
```

Comment:
Hello @odoo/rd-security :wave: 

This PR is not yet ready for review but I wanted to ask what you advise in that case.

A bit of context: This PR allows skipping all the steps of the checkout flow from the cart to the confirmation page by using an express checkout button 'Apple Pay' or 'Google Pay'. Stripe processes the Payment. For Apple Pay to work, the customer needs to register his domain name. Stripe handles that through an API but calls back our server to check if the URL works. The URL called by Stripe needs to return the string contained in [apple-developer-merchantid-domain-association](https://github.com/odoo/odoo/blob/ccd7f8fee5ad460d7d2400e62b34f91166b7350c/addons/website_payment_stripe/static/apple-developer-merchantid-domain-association).

Is there a better (a more secure?) way than the `open()` method to do that?

See also: https://stripe.com/docs/stripe-js/elements/payment-request-button?html-or-react=html#verifying-your-domain-with-apple-pay

Have a nice weekend :sun_with_face:

Reply:
> Is there a better (a more secure?) way than the `open()` method to do that?

Yes, since you're fetching from module static data you should use the `file_open` helper function, ideally with an extension filter (though the filter feature doesn't work with no extension so not an option here).

Incidentally using `file_open` means you don't need to use `get_resource_path`, it does the normalisation internally, so you can just give it the path `"/website_payment_stripe/static/apple-developer-merchantid-domain-association"` and it should